src/lib/components/wizard-navigation-bar.component.vertical.less

```diff
@@ -31,368 +31,191 @@
 
-        .step-indicator {
-          .state-circle(@small-dot-width, @small-dot-height, 0);
-          .state-circle-with-background(@wz-color-default);
-        }
+.state-circle-with-border(@color) {
+  border-width: @border;
+  border-style: solid;
+  border-color: @color;
+}
+.state-circle-with-border-hover(@color) {
+  border-color: darken(@color, 10%);
+}
+.state-circle-with-border-nohover(@color) {
+  border-color: @color;
+}
 
-        a:hover .step-indicator {
-          .state-circle-hover(@small-dot-width, @small-dot-height, 0);
-          .state-circle-with-background-hover(@wz-color-default);
-        }
+.state-circle-with-border-and-content(@color) {
+  .state-circle-with-border(@arguments);
+  color: @color;
+}
+.state-circle-with-border-and-content-hover(@color) {
+  .state-circle-with-border-hover(@arguments);
+  color: darken(@color, 10%);
+}
+.state-circle-with-border-and-content-nohover(@color) {
+  .state-circle-with-border-nohover(@arguments);
+  color: @color;
+}
 
-        div {
-          min-height: @small-dot-height;
-        }
-      }
+.state-circle-with-background(@color) {
+  background-color: @color;
+}
+.state-circle-with-background-hover(@color) {
+  background-color: darken(@color, 5%);
+}
+.state-circle-with-background-nohover(@color) {
+  background-color: @color;
+}
 
-      li.current .step-indicator {
-        .state-circle-with-background(@wz-color-current);
-      }
-      li.done .step-indicator {
-        .state-circle-with-background(@wz-color-done);
-      }
-      li.optional .step-indicator {
-        .state-circle-with-background(@wz-color-optional);
-      }
-      li.editing .step-indicator {
-        .state-circle-with-background(@wz-color-editing);
-      }
+.state-circle-with-background-and-content(@color) {
+  .state-circle-with-background(@color);
+  color: black;
+}
+.state-circle-with-background-and-content-hover(@color) {
+  .state-circle-with-background-hover(@color);
+}
+.state-circle-with-background-and-content-nohover(@color) {
+  .state-circle-with-background-nohover(@color);
+}
 
-      li.current a:hover .step-indicator {
-        .state-circle-with-background-hover(@wz-color-current);
-      }
-      li.done a:hover .step-indicator {
-        .state-circle-with-background-hover(@wz-color-done);
-      }
-      li.optional a:hover .step-indicator {
-        .state-circle-with-background-hover(@wz-color-optional);
-      }
-      li.editing a:hover .step-indicator {
-        .state-circle-with-background-hover(@wz-color-editing);
-      }
 
-      // default steps shouldn't change when hovered, because they aren't clickable
-      li.default a:hover .step-indicator {
-        .state-circle-with-background-nohover(@wz-color-current);
-      }
+// Helper mixin to define a step indicator stype like 'small', 'large-filled-symbols' etc.
+//
+// Arguments:
+//   @width  - step indicator width
+//   @height - step indicator height
+//   @border - step indicator border
```

Comment:
I think the arguments need some more description.
For example, the `@border` argument description doesn't specify what information the argument needs to contain:
- whether a border should be drawn (i.e. a boolean)
- how thick a border should be drawn 
- or something entirely different?

Reply:
bfa1d631